### DISCUSSION TITLE
Command manager: is it possible to undo specific command?

### DISCUSSION THREAD

**Discussion by gintarasgraciovas** [clientIO/joint#2617] [Q&A] [answered]
### Introduction

I need to undo changes made to a graph when errors are received from the backend (BE). Here's my scenario:

- A user with a slow internet connection attempts to add two connections to the graph.

- The user receives an error for the first connection and subsequently for the second one as well.

In response to these errors, I need to first undo the changes made by the first connection and then address the second one.

Is it possible to do something like that?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
No, you cannot use the command manager this way. It's not possible to undo a specific command because one operation can affect subsequent operations.

For instance, consider an `add` operation followed by a `change:color` operation. If you undo the `add` operation, it removes the associated node. However, the `change:color` operation remains in the undo stack, referencing a node that no longer exists.

This scenario highlights the complexity that can be addressed by [operational transformation](https://en.wikipedia.org/wiki/Operational_transformation) (OT) techniques. Although OT provides a general solution, if your specific requirement is limited to undoing link reconnection operations, a simpler solution may suffice.

**Reply by gintarasgraciovas** [reply]
Thanks for the explanation.
